snoop/data/s3.py

```diff
@@ -278,9 +408,8 @@ def target_is_mounted(path):
     )
 
 
-def adjust_s3_mounts(mount_name, old_info,
-                     bucket, mount_mode,
-                     password_file_path, address, target_path, logfile_path):
+def adjust_mounts(mount_name, old_info,
+                  mount_func, mount_args):
```

Comment:
Add new boolean argument `check_pid` that is True on S3 and False on webdav.

Then, at line 434, replace
```
new_info = _clear_dead(old_info)
```

with

```
if check_pid:
    new_info = _clear_dead(old_info)
```


This should fix the problem with non-foreground mounting.